didman/didman.go

```diff
@@ -87,6 +114,9 @@ func (d *didman) Name() string {
 }
 
 func (d *didman) AddEndpoint(id did.DID, serviceType string, u url.URL) (*did.Service, error) {
+	unlockFn := d.callSerializer.Lock(id.String())
```

Comment:
I think you can even make this a one-liner

Reply:
Yes, but that will be unreadable:
`defer d.callSerializer.Lock(id.String())()`

Reply:
I can read that 🤣